### ISSUE TITLE
Some questions regarding the loss function and other aspects in the article

### ISSUE THREAD

**Issue by mourad1081** [WaqasSultani/AnomalyDetectionCVPR2018#40]
I have never seen a loss function which compare positive examples and negative examples. All the loss function I have seen so far (i'm still a student) compare the true value of _y_ with the estimated value of _y_.

I am a bit confused, thus I have some questions : 

**(I will re-read the paper again and delete some questions if I find an anwser.)**

- Is it a common way to implement loss functions like yours (by comparing a positive example with a negative one)? Why is it more efficient than simply comparing the true value of y with the estimated value of y as in classical statistical methods.

- Is there a technical term for that kind of loss function? I have searched on Google about "ranking learning" and found a book called "Learning to rank". Am I searching in the right direction?

- You say in the article: _"Recently, deep ranking networks have been used in several computer vision applications and have shown state-of-the-art performances."_ I have never encountered the notion of "deep ranking networks" in my degree. Does that notion mean _"A deep neural network where the los function gives a ranking score."_ ? If not, what is it then? When I search on Google, the only thing I find is a paper called _Learning Fine-grained Image Similarity with Deep Ranking_ but no Wiki page, nor any webpage explaining what is a ranking network.

**Comment by AmmarKamoona** [comment]
"I have never seen a loss function which compares positive examples and negative examples"
Hi mourad1081, 
what do you mean by comparing positive examples with negative examples? Do you mean like comparing Positive C3D features with negative C3D features examples?
